### PR TITLE
Updates for GET and POST

### PR DESCRIPTION
**Pull request by 311chaos**
- updated to handle the events for _handleAdd and _handleAll, these now get tossed back to the main component to deal with the _images array.
- Passing in the data for addImage using FormData()
- Updated the demo page to allow for mocking some data (there is some weirdness if you have already added images through the API)
- Updated tests

### CODE REVIEW

**Comment by 311chaos** [comment]
I will keep this open until I have added in the DELETE and the POST and done a little refactoring.

**Comment by zack37** [comment]
Uploading image in the demo doesn't work

File: ll-property-images.html [hunk 1]
```diff
@@ -25,32 +25,38 @@
...
-          img-id="{{item.id}}"
-          name="{{item.name}}"
+          id="{{item._id}}"
+          img-id="{{item._id}}"
+          name="{{item.fileName}}"
           title="{{item.title}}"
           description="{{item.description}}"
           src="{{item.url}}"
-          is-default-image$="{{item.isDefaultImage}}"
+          is-default$="{{item.isDefault}}"
```
Comment (zack37) [inline ll-property-images.html]: The $= maybe be unnecessary since it's for attributes, not properties.

**Comment by zack37** [comment]
Mostly looks good. I'm still having trouble uploading in the demo, but we can talk about that later. One thing that would be nice to add is documentation for events fired, specifically for not the manager. I see the manager has events documented.

**Comment by 311chaos** [comment]
It's because the images Api was changed to "images-api" but is not working in prod anymore. (Under images or images-api).

We need to sync up with Colton and Jason to see if that is pending another PR.

Sent from Outlookhttp://aka.ms/Ox5hz3

On Wed, Sep 2, 2015 at 7:11 AM -0700, "Zack Smith" <notifications@github.com<mailto:notifications@github.com>> wrote:

Mostly looks good. I'm still having trouble uploading in the demo, but we can talk about that later. One thing that would be nice to add is documentation for events fired.

##

**Comment by 311chaos** [comment]
@zack37 - Updates made. Can you leave this branch open after the merge?

**Pull request merged**